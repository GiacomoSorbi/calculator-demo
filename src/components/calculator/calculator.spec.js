import * as React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Calculator from './'

describe('Calculator should', () => {
  it('exist', () => {
    expect(Calculator).toBeDefined()
  })

  it('have a consistent structure', () => {
    const wrapper = mount(<Calculator />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
