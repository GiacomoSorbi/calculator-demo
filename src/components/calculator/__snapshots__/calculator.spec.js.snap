// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should have a consistent structure 1`] = `
<Calculator>
  <div
    className="calculator-container"
  >
    <Display
      displayedText="3.14"
    >
      <div
        className="display-container"
      >
        <span>
          3.14
        </span>
      </div>
    </Display>
    <div
      className="calculator-buttons"
    >
      <Button
        key="("
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          (
        </button>
      </Button>
      <Button
        key="CE"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          CE
        </button>
      </Button>
      <Button
        key=")"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          )
        </button>
      </Button>
      <Button
        key="C"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          C
        </button>
      </Button>
      <Button
        key="7"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          7
        </button>
      </Button>
      <Button
        key="8"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          8
        </button>
      </Button>
      <Button
        key="9"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          9
        </button>
      </Button>
      <Button
        key="/"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          /
        </button>
      </Button>
      <Button
        key="4"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          4
        </button>
      </Button>
      <Button
        key="5"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          5
        </button>
      </Button>
      <Button
        key="6"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          6
        </button>
      </Button>
      <Button
        key="*"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          *
        </button>
      </Button>
      <Button
        key="1"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          1
        </button>
      </Button>
      <Button
        key="2"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          2
        </button>
      </Button>
      <Button
        key="3"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          3
        </button>
      </Button>
      <Button
        key="-"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          -
        </button>
      </Button>
      <Button
        key="."
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          .
        </button>
      </Button>
      <Button
        key="0"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          0
        </button>
      </Button>
      <Button
        key="="
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          =
        </button>
      </Button>
      <Button
        key="+"
        onClick={[Function]}
      >
        <button
          className="calculator-btn"
          onClick={[Function]}
        >
          +
        </button>
      </Button>
    </div>
  </div>
</Calculator>
`;
